Assets/MixedRealityToolkit-SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -0,0 +1,489 @@
+                {
+                    GameObject.Find("DebugLog").GetComponent<TextMesh>().text = eventData.MixedRealityInputAction.Description + " " +
+                                                                            Time.unscaledDeltaTime.ToString();
+                }
+                else if (m_handsPressedInputSourceMap.Count == 1)
+                {
+                    GameObject.Find("DebugOut").GetComponent<TextMesh>().text = eventData.MixedRealityInputAction.Description + " " +
+                                                                            Time.unscaledDeltaTime.ToString();
+                }
+
+                GameObject.Find("DebugText").GetComponent<TextMesh>().text = eventData.SourceId.ToString();
+            }
+
+            Vector3 inputPosition = Vector3.zero;
+            m_handsPressedInputSourceMap[eventData.SourceId] = eventData.InputSource;
+
+            // Add to hand map
+            if (eventData.InputSource.SourceName.Contains("Hand") == true)
+            {
+                inputPosition = m_handsPoseMap[eventData.SourceId];
+            }
+            else
+            {
+                eventData.InputSource.Pointers[0].TryGetPointerPosition(out inputPosition);   
+            }
+            m_handsPressedLocationsMap[eventData.SourceId] = inputPosition;
+
+            UpdateStateMachine();
+            eventData.Use();
+        }
+
+        /// <summary>
+        /// Event Handler receives input from inputSource
+        /// </summary>
+        public void OnInputUp(InputEventData eventData)
+        {
+            RemoveSourceIdFromHandMap(eventData.SourceId);
+            UpdateStateMachine();
+            eventData.Use();
+        }
+
+        /// <summary>
+        /// from IMixedRealitySpacialInputHandler
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnPoseInputChanged(InputEventData<MixedRealityPose> eventData)
+        {
+            if (debugText)
+            {
+           //    debugText.text = "On pose changed " + eventData.MixedRealityInputAction.Description + " " + Time.unscaledDeltaTime.ToString();
+            }
+            if (eventData.InputSource.SourceName.Contains("Hand"))
+            {
+                m_handsPoseMap[eventData.SourceId] = eventData.InputData.Position;
+            }
+        }
+
+        /// <summary>
+        /// OnSourceLost
+        /// </summary>
+        public void OnSourceLost(SourceStateEventData eventData)
+        {
+            RemoveSourceIdFromHandMap(eventData.SourceId);
+            UpdateStateMachine();
+            eventData.Use();
+        }
+        #endregion Event Handlers
+
+        #region Unused Event Handlers
+        /// <summary>
+        /// OnSourceDetected Event Handler
+        /// </summary>
+        public void OnSourceDetected(SourceStateEventData eventData) { }
+        public void OnInputPressed(InputEventData<float> eventData)
```

Comment:
minor formatting suggestion, please format all empty implementation event methods similar to OnSourceDetected.

thanks!

Reply:
done